tests/unit/test_retriever.py

```
@@ -0,0 +1,71 @@
+class _SpyStore(MemoryStore):
+    """Minimal MemoryStore that records query calls and returns canned results."""
+
+    def __init__(self, results: list[dict]) -> None:
+        self._results = results
+        self.query_calls: list[tuple[list[float], int]] = []
+
+    def query(self, embedding: list[float], top_k: int = 5) -> list[dict]:
+        self.query_calls.append((list(embedding), top_k))
+        return list(self._results)
+
+    def add(self, id, text, embedding, metadata): ...  # noqa: E704
```

Ruff (0.15.10)

[error] Function argument `id` is shadowing a Python builtin

(A002)